contracts/libraries/LibTaskModule.sol

```diff
@@ -0,0 +1,192 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.12;
+
+import {_call, _delegateCall} from "../functions/FExec.sol";
+import {LibDataTypes} from "./LibDataTypes.sol";
+import {ITaskModule} from "../interfaces/ITaskModule.sol";
+
+/**
+ * @notice Library to call task modules on task creation and execution.
+ */
+library LibTaskModule {
+    /**
+     * @notice Delegate calls task modules on create task to initialise them.
+     *
+     * @param _taskId Unique hash of the task. {See LibTaskId-getTaskId}
+     * @param _taskCreator The address which created the task.
+     * @param _execAddress Address of contract that will be called by Gelato.
+     * @param _execData Execution data to be called with / function selector.
+     * @param _moduleData Modules that will be used for the task. {See LibDataTypes-ModuleData}
+     * @param taskModuleAddresses The storage reference to the mapping of modules to their address.
+     */
+    function onCreateTask(
+        bytes32 _taskId,
+        address _taskCreator,
+        address _execAddress,
+        bytes memory _execData,
+        LibDataTypes.ModuleData memory _moduleData,
+        mapping(LibDataTypes.Module => address) storage taskModuleAddresses
+    ) internal {
+        uint256 length = _moduleData.modules.length;
+
+        _validModules(length, _moduleData.modules);
+
+        for (uint256 i; i < length; i++) {
+            LibDataTypes.Module module = _moduleData.modules[i];
+
+            address taskModuleAddress = taskModuleAddresses[module];
```

Comment:
we could reduce the gas overhead by passing moduleAddresses rather than Enums which then get mapped to modules. However, this should only be done if the gas savings are worth the additional "overhead" of having to manage the module addresses off-chain. Lets discuss. One thing to consider here: Certain projects are starting to parallelize their tasks for performance boost. Enabling them to have a lower "task creation" overhead might make it worth it.